demisto_sdk/commands/validate/validators/BC_validators/BC104_have_commands_or_args_name_changed.py

```diff
@@ -0,0 +1,85 @@
+    related_field = "name"  # TODO - what is the field name?
+    is_auto_fixable = False
+    expected_git_statuses = [GitStatuses.MODIFIED]
+    related_file_type = [RelatedFileType.YML]
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        results: List[ValidationResult] = []
+        for content_item in content_items:
+            old_content_item = content_item.old_base_content_object
+
+            # commands name changed
+            new_commands_names = [command.name for command in content_item.commands]
+            old_commands_names = [command.name for command in old_content_item.commands]  # type: ignore
+
+            commands_diff = compare_lists(
+                sub_list=old_commands_names, main_list=new_commands_names
+            )
+            if commands_diff:
+                results.append(
+                    ValidationResult(
+                        validator=self,
+                        message=self.error_message.format(
+                            unique_message=f"to the names of the following existing commands: {', '.join(commands_diff)}.",
+                        ),
+                        content_object=content_item,
+                    )
+                )
+
+            # arguments name changed
+            args_diff_per_command_summary = []
+            for command in content_item.old_base_content_object.commands:  # type: ignore
+                new_args_per_command = []
+                old_args_per_command = [argument.name for argument in command.args]
+                current_command_name = command.name
+
+                find_new_command = find_command(
+                    content_item.commands, current_command_name
+                )
+                new_args_per_command = (
+                    [argument.name for argument in find_new_command.args]
+                    if find_new_command
+                    else []
+                )
+
+                if new_args_per_command:
+                    diff_per_command = compare_lists(
+                        sub_list=old_args_per_command, main_list=new_args_per_command
+                    )
+                    if diff_per_command:
+                        args_diff_per_command_summary.append(
+                            f"In command '{current_command_name}' the following arguments have been changed: {', '.join(diff_per_command)}."
+                        )
+            if args_diff_per_command_summary:
+                results.append(
+                    ValidationResult(
+                        validator=self,
+                        message=self.error_message.format(
+                            unique_message=f"to the names of existing arguments: {' '.join(args_diff_per_command_summary)}",
+                        ),
+                        content_object=content_item,
+                    )
+                )
```

Comment:
I don't like the idea of returning 2 messages from the same validation.
Let's either find a way to return one validationResult, or split this validation into two validations.

Reply:
fixed